python/pecos/qeclib/steane/qec/qec_3parallel.py

```diff
@@ -48,3 +49,5 @@ def __init__(
             FlagLookupQASMActiveCorrectionX(q, syn_x, syndromes, last_raw_syn_x, pf_z, flag_x, flags, scratch),
             FlagLookupQASMActiveCorrectionZ(q, syn_z, syndromes, last_raw_syn_z, pf_x, flag_z, flags, scratch),
         )
+        if flag_bit is not None:
+            self.extend(If(flags != 0).Then(flag_bit.set(1)))
```

Comment:
I don't think extra information should be extracted from lower level operations. To prevent the accumulation of arguments, it is most preferable for this info to be extracted at the user level. Next, best would be the class operations. Perhaps the best compromise might be to expose flag/syndrome information to the user and have them if off of that information.

Reply:
Maybe I need to think about making variables obviously private using underscores and provide getter methods to expose useful info publicly to make it easier for users.

Reply:
I am not sure what you mean by "lower level operations", but as you suggest what I am really trying to do here is expose flag/syndrome info.  My thinking was to give the user the reigns here because it seems to me like the `Steane` class cannot know a priori how many QEC cycles are in a program, and therefore how many syndrome bits to save (or even whether to save them).